[BUG] FT_MOTION compilation error

### Did you test the latest `bugfix-2.1.x` code?

Yes, and the problem still exists.

### Bug Description

I have error when compile marlin with ft_motion:
```
Marlin/src/module/ft_motion.cpp:97:9: error: either all initializer clauses should be designated or none of them should be
   97 |       , x:{ .ena = false, .d_zi = { 0.0f }, .Ai = { 0.0f }, .Ni = { 0 }, .max_i = 0 } // ena, d_zi[], Ai[], Ni[], max_i
```

We can either assign initializers to everything or get rid of them

Possible solution 1:
```
#if HAS_FTM_SHAPING
  FTMotion::shaping_t FTMotion::shaping = {
    zi_idx: 0
    #if HAS_X_AXIS
      , x:{ false, { 0.0f }, { 0.0f }, { 0 }, 0 } // ena, d_zi[], Ai[], Ni[], max_i
    #endif
    #if HAS_Y_AXIS
      , y:{ false, { 0.0f }, { 0.0f }, { 0 }, 0 } // ena, d_zi[], Ai[], Ni[], max_i
    #endif
  };
#endif
```

Possible solution 2:
 ```
#if HAS_FTM_SHAPING
  FTMotion::shaping_t FTMotion::shaping = {
    0
    #if HAS_X_AXIS
      , { false, { 0.0f }, { 0.0f }, { 0 }, 0 } // ena, d_zi[], Ai[], Ni[], max_i
    #endif
    #if HAS_Y_AXIS
      , { false, { 0.0f }, { 0.0f }, { 0 }, 0 } // ena, d_zi[], Ai[], Ni[], max_i
    #endif
  };
#endif
```

### Bug Timeline

_No response_

### Expected behavior

_No response_

### Actual behavior

_No response_

### Steps to Reproduce

_No response_

### Version of Marlin Firmware

bugfix-2.1.x 10/11/2024

### Printer model

any

### Electronics

MKS_SGEN_L LPC1768

### LCD/Controller

_No response_

### Other add-ons

_No response_

### Bed Leveling

None

### Your Slicer

None

### Host Software

None

### Don't forget to include

- [X] A ZIP file containing your `Configuration.h` and `Configuration_adv.h`.

### Additional information & file uploads

[config.zip](https://github.com/user-attachments/files/17342954/config.zip)

--- a/Marlin/src/module/ft_motion.cpp
+++ b/Marlin/src/module/ft_motion.cpp
@@ -92,7 +92,7 @@ uint32_t FTMotion::interpIdx = 0;               // Index of current data point b
 // Shaping variables.
 #if HAS_FTM_SHAPING
   FTMotion::shaping_t FTMotion::shaping = {
-    0
+    zi_idx: 0
     #if HAS_X_AXIS
       , x:{ false, { 0.0f }, { 0.0f }, { 0 }, 0 } // ena, d_zi[], Ai[], Ni[], max_i
     #endif

